src/Test/L0/Plugin/TestFilePublisher/TestFilePublisherLogPluginL0.cs

```diff
@@ -311,8 +311,10 @@ public async Task TestFilePublisherLogPlugin_EnableForBuildPipeline()
             var testFilePublisher = new Mock<ITestFilePublisher>();
             var logger = new Mock<ITraceLogger>();
             var telemetry = new Mock<ITelemetryDataCollector>();
+            var telemetryProps = new Dictionary<string, Object>();
 
             telemetry.Setup(x => x.PublishCumulativeTelemetryAsync()).Returns(Task.FromResult(TaskResult.Succeeded));
+            telemetry.Setup(x => x.PublishTelemetryAsync(It.IsAny<string>(), It.IsAny<Dictionary<string, Object>>())).Callback<string, Dictionary<string, Object>>((feature, props) => telemetryProps = props).Returns(Task.FromResult(TaskResult.Succeeded));
```

Comment:
>elemetryProps = props [](start = 184, length = 21)

instead of assigning this here can you call a utility function called AssertPipelineData and assert the props inside that so that it is re-usable?